lib/bolt/config.rb

```diff
@@ -82,6 +88,16 @@ def initialize(boltdir, config_data, overrides = {})
         @transports[key] = transport.default_options
       end
 
+      default_config_data = {}
```

Comment:
As a rule we don't read file in `initialize` since it makes it impossible to avoid system config during tests and boltspec functionality.

Reply:
I think this would have to be a class method that get's called by the other class methods before new.